pages/_app/index.jsx

```diff
@@ -121,6 +131,10 @@ class App extends NextApp {
   componentWillUnmount() {
     const { router } = this.props
     router.events.off('routeChangeComplete', this.handleRouteChange)
+    window.removeEventListener(
+      'unhandledrejection',
+      this.handlePromiseRejection
+    )
```

Comment:
Why do you need capture it separately? Does not `componentDidCatch` handle everything?

Reply:
Not it doesn't. See here: https://github.com/facebook/react/issues/11334#issuecomment-338656383

Actually I think it's useless to have check incomponentDidCatch. I cannot think about the use case for that.